Upload artifacts example (#50)
ddev/github-action-add-on-test: README.md

For more information on `bats` tags and filtering tests by tags, refer to the [bats documentation](https://bats-core.readthedocs.io/en/stable/writing-tests.html#tagging-tests).

#### Preserving Artifacts

> GitHub Actions provides a UI for browsing test artifacts. Configure your job as shown below:

Reviewer: ```suggestion
GitHub Actions provides a UI for browsing test artifacts. To use it, store `$TESTDIR` in `$GITHUB_ENV` within your Bats tests (see [example](https://github.com/ddev/ddev-addon-template/blob/main/tests/test.bats)), and configure your job as follows:
```

```yaml
jobs:
  tests:
    steps:
      ...

      - name: Upload artifacts
        uses: actions/upload-artifact@v4
        with:

>           name: artifact-name

Reviewer: Add unique name:

```suggestion
          name: artifact-name-${{ matrix.ddev_version }}
```

          path: |

>           ${{ env.TESTDIR }}/path/to/file

Reviewer: Noticed wrong indentation:
```suggestion
            ${{ env.TESTDIR }}/path/to/file
```

>             ${{ env.TESTDIR }}/path/to/file

Reviewer: ```suggestion
            ${{ env.TESTDIR }}/path/to/file/or/directory
```

>       - name: Upload artifacts
>         uses: actions/upload-artifact@v4
>         with:
>           name: artifact-name-${{ matrix.ddev_version }}
>           path: |
>             ${{ env.TESTDIR }}/path/to/file/or/directory

Reviewer: ```suggestion
      # Optionally, you may need to zip some artifacts before uploading them
      # The path for one of the files in artifact is not valid
      # - name: Zip artifacts that can't be processed by actions/upload-artifact
      #   if: always()
      #   run: |
      #     for d in ${{ env.TESTDIR }}*/; do
      #       if [ -d "$d/path/to/directory" ]; then
      #         (cd "$d/path/to" && zip -r directory.zip directory)
      #       fi
      #     done

      - name: Upload artifacts
        if: always()
        uses: actions/upload-artifact@v4
        with:
          name: artifact-${{ matrix.ddev_version }}
          path: |
            ${{ env.TESTDIR }}*/tests
            ${{ env.TESTDIR }}*/web/*.junit.xml
            ${{ env.TESTDIR }}*/path/to/directory.zip
```
